### PR TITLE
PROC-735 JSON Schema Draft 7 Support

### PR DESCRIPTION
**Pull request by michaelcarter-wf**
## Ultimate problem:
- We don't support draft 7.

## How it was fixed:
- Run the draft 7 tests.

## Potential areas of regression:

---

> __FYA:__ @michaelcarter-wf

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 Recommendation: **:warning: Minor version bump**


```diff
@@ line 74: package:json_schema/src/json_schema/constants.dart @@
class SchemaVersion implements Comparable<SchemaVersion>
+     static SchemaVersion get draft7
//    Adding a field is a minor change.

```

Showing results for e1a7c36bb371fa51b9cdeab6bc1164e05cd02e12
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.1
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/json_schema/e1a7c36bb371fa51b9cdeab6bc1164e05cd02e12).

File: lib/src/json_schema/validator.dart [hunk 1]
```diff
@@ -517,6 +573,32 @@ class Validator {
...
+        if (schema.thenSchema == null) return true;
+        if (!Validator(schema.thenSchema).validate(instance)) {
+          _err('${schema.path}/then: then violated ($instance, ${schema.thenSchema})', instance.path,
+              schema.path + '/then');
+        }
+      } else {
+        // Bail out early if no "else" is specified.
+        if (schema.elseSchema == null) return true;
+        if (!Validator(schema.elseSchema).validate(instance)) {
+          _err('${schema.path}/else: then violated ($instance, ${schema.elseSchema})', instance.path,
```
Comment (jordanross-wf) [inline lib/src/json_schema/validator.dart:592]:
```suggestion
          _err('${schema.path}/else: else violated ($instance, ${schema.elseSchema})', instance.path,
```

**Review by jordanross-wf — APPROVED** [review]

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**